### ISSUE TITLE
synchronization bug (create -> start -> kill)

### ISSUE THREAD

**Issue by devimc** [01org/cc-oci-runtime#828]
running container with runc
```
for i in `seq 1 15`; do docker run --runtime=runc -dti --name bla$i ubuntu bash -c "trap \"exit 15\" SIGUSR1 ; while : ; do sleep 1; done" && docker kill -s SIGUSR1 bla$i ; done
```
the status of all container will be ```Exited (15)```

but using cor
```
for i in `seq 1 15`; do docker run --runtime=cor -dti --name bla$i ubuntu bash -c "trap \"exit 15\" SIGUSR1 ; while : ; do sleep 1; done" && docker kill -s SIGUSR1 bla$i ; done
```
the status of some containers will be 0

**Comment by grahamwhaley** [comment]
Hi @devimc Do you know if this exists still for CC3.x https://github.com/clearcontainers/runtime ?  I just did a test run, and did not see the issue - would be good if somebody else could verify?
btw, @chavafg - do we test this in the QA tests? Maybe we can add a test if we don't have one already?

I ask as we still reference this issue in https://github.com/clearcontainers/tests/pull/381

**Comment by chavafg** [comment]
Still fails for me using 2nd test of: https://github.com/01org/cc-oci-runtime/blob/master/tests/integration/docker/kill.bats 

We still do not have this test on the `clearcontainers/tests` repo.

I tested this under a VM... maybe a race condition? @grahamwhaley are you running on baremetal?

**Comment by grahamwhaley** [comment]
@chavafg yep, tested on a bare metal Ubuntu, just one run of the simple script at the top of this Issue.
If we have no record of this in CC3.x repos, can we at minimum open an Issue there pls?

**Comment by chavafg** [comment]
@grahamwhaley, created https://github.com/clearcontainers/runtime/issues/430

**Comment by devimc** [comment]
Hi @grahamwhaley , I don't think so
